File: Live2D/V2/Core/ParamDefSet.cpp
#include "ParamDefSet.hpp"
#include "ParamDefFloat.hpp"
#include "BinaryReader.hpp"

namespace live2d {

ParamDefSet::~ParamDefSet() {
    for (auto* p : mParamDefList) delete p;
}

void ParamDefSet::read(BinaryReader& br) {
    mParamDefList = br.readObject<std::vector<ParamDefFloat*>>();
}

} // namespace live2d

File: package/live2d/v2/core/param/param_def_set.py
from ..io.iserializable import ISerializable
class ParamDefSet(ISerializable):

    def __init__(self):
        self.paramDefList = None

    def getParamDefFloatList(self):
        return self.paramDefList

    def read(self, br):
        self.paramDefList = br.readObject()
